src/System.CommandLine.Tests/Binding/ModelBindingCommandHandlerTests.cs

```diff
@@ -458,6 +458,114 @@ public async Task When_binding_fails_due_to_parameter_naming_mismatch_then_handl
             c.AssertBoundValue(boundValue);
         }
 
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(bool))]
+        [InlineData(typeof(ClassWithSetter<int>))]
+        [InlineData(typeof(ClassWithCtorParameter<string>))]
+        [InlineData(typeof(ClassWithSetter<string>))]
+        [InlineData(typeof(FileInfo))]
+        [InlineData(typeof(FileInfo[]))]
+        [InlineData(typeof(string[]))]
+        [InlineData(typeof(List<string>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(List<int>))]
+        public async Task Handler_method_receives_command_arguments_explicitly_bound_to_the_specified_type(
+            Type type)
+        {
+            var c = BindingCases[type];
+
+            var captureMethod = GetType()
+                                .GetMethod(nameof(CaptureMethod), BindingFlags.NonPublic | BindingFlags.Static)
+                                .MakeGenericMethod(c.ParameterType);
+            var parameter = captureMethod.GetParameters().First();
+
+            var handler = CommandHandler.Create(captureMethod);
+
+            var argument = new Argument
+            {
+                Name = "value",
+                ArgumentType = c.ParameterType
+            };
+
+            var command = new Command(
+                "command")
+            {
+                argument
+            };
+            handler.BindParameter(parameter, argument);
+            command.Handler = handler;
+
+            var parseResult = command.Parse(c.CommandLine);
+
+            var invocationContext = new InvocationContext(parseResult);
+
+            await handler.InvokeAsync(invocationContext);
+
+            var boundValue = ((BoundValueCapturer)invocationContext.InvocationResult).BoundValue;
+
+            boundValue.Should().BeOfType(c.ParameterType);
+
+            c.AssertBoundValue(boundValue);
+        }
+
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(bool))]
+        [InlineData(typeof(ClassWithCtorParameter<int>))]
+        [InlineData(typeof(ClassWithSetter<int>))]
+        [InlineData(typeof(ClassWithCtorParameter<string>))]
+        [InlineData(typeof(ClassWithSetter<string>))]
+        [InlineData(typeof(FileInfo))]
+        [InlineData(typeof(FileInfo[]))]
+        [InlineData(typeof(string[]))]
+        [InlineData(typeof(List<string>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(List<int>))]
+        public async Task Handler_method_receives_command_options_explicitly_bound_to_the_specified_type(
```

Comment:
I think this should be named `Handler_method_receive_option_arguments_explicitly_bound_to_the_specified_type`